### DISCUSSION TITLE
There is sparse vector implementation in River?

### DISCUSSION THREAD

**Discussion by giturra** [online-ml/river#694] [Q&A] [answered]
Hi everyone!

First I want to thank you for this excellent package.

I have to replicate some features of software made in [MOA.](https://moa.cms.waikato.ac.nz/) Some class of the package of the software called ["SparseInstance"](https://weka.sourceforge.io/doc.dev/weka/core/SparseInstance.html) instance, which is a sparse vector implementation of MOA. I was reading the River documentation, but I couldn't find any similar. So I'm not sure if it exists under another name.

I'm a little curious about the [VectorDict](https://riverml.xyz/latest/api/utils/VectorDict/) class that I could use for something similar.

Thanks, everyone :)

**Comment by MaxHalford** [comment, marked as answer]
Well, in River we use `dict`s, which are sparse by design. They don't have a predetermined length. A `VectorDict` is "just" a fancy `dict` that is optimized for algebric operations.
